pg_search/src/api/operator.rs

```diff
@@ -449,6 +489,48 @@ pub unsafe fn attname_from_var(
     (heaprelid, attname)
 }
 
+/// Given a [`pg_sys::PlannerInfo`] and a [`pg_sys::Node`] from it, figure out the name of the `Node`,
+/// it supports `FuncExpr` and `Var` nodes. Note that for the heap relation, the `Var` must be
+/// the first argument of the `FuncExpr`.
+///
+/// # Return
+///
+/// Returns the heap relation [`pg_sys::Oid`] that contains the `Node` along with its name.
+pub unsafe fn attname_from_node(
+    root: *mut pg_sys::PlannerInfo,
+    node: *mut pg_sys::Node,
+) -> Option<(pg_sys::Oid, Option<String>)> {
+    match (*node).type_ {
+        NodeTag::T_FuncExpr => {
+            // We expect the funcexpr to contain the var of the field name we're looking for
+            let funcexpr = nodecast!(FuncExpr, T_FuncExpr, node).expect("node is not a FuncExpr");
+            let heaprelid = PgList::<pg_sys::Node>::from_pg((*funcexpr).args)
+                .iter_ptr()
+                .find_map(|arg| {
+                    if is_a(arg, pg_sys::NodeTag::T_Var) {
+                        let var = nodecast!(Var, T_Var, arg).expect("node is not a Var");
+                        let (heaprelid, _, _) = find_var_relation(var, root);
+                        Some(heaprelid)
+                    } else {
+                        None
+                    }
+                })?;
+            let indexrel =
+                locate_bm25_index(heaprelid).expect("could not find bm25 index for heaprelid");
+
+            let attnum = find_funcexpr_attnum(&indexrel, node)?;
```

Comment:
I think that this additional requirement is fine (that the `Node` must only be referring to a `bm25` indexed relation), but possibly worth noting in the name or docstring of the function, since otherwise it seems generic to any heap relation.